### PR TITLE
Provide better custom logger example in README

### PR DESCRIPTION
**Pull request by ryanwalls**
## What was changed

Updated the README with a more helpful logging example.

## Why?

The current example did not make it clear how to "feed the logger" into Temporal.  Grabbed the better example from https://docs.temporal.io/go/how-to-set-clientoptions-in-go/#logger.

## Checklist

1. Closes 
n/a

2. How was this tested:

n/a

3. Any docs updates needed?

Done

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,13 +25,19 @@ Although the Go SDK does not support most third-party logging solutions natively
 package main
 import (
 	"github.com/sirupsen/logrus"
+	"go.temporal.io/sdk/client"
```
Comment (cretz) [inline README.md:28]: I wonder if we should just remove this section (notice there aren't other examples/docs here in the README) and that way people will look at https://github.com/temporalio/samples-go instead (specifically https://github.com/temporalio/samples-go/tree/main/zapadapter).

Thoughts?

**Review by cretz — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (cretz) [reply README.md:28]: > and a very austere README may send the wrong message.

The README is already very tiny compared to docs.

> Perhaps some of the sections can be replaced with links to other docs, but keep the section titles?

There aren't really any sections of note besides this one.

I can merge this and we can just table the idea of altering the README more drastically. Sound good?

Comment (ryanwalls) [reply README.md:28]: Merging for now is fine.  Or you can just replace the one section we are talking about with a link to the docs for the logger.  I'm good either way.

Comment (cretz) [reply README.md:28]: Can merge. Can you update the branch w/ master? and get CI passing?

**Comment by cretz** [comment]
Merged, thanks!

**Pull request merged**